internal/gatewayapi/route.go

```diff
@@ -314,12 +314,33 @@ func (t *Translator) processHTTPRouteRule(httpRoute *HTTPRouteContext, ruleIdx i
 		ruleRoutes = append(ruleRoutes, irRoute)
 	}
 
+	var sessionPersistence *ir.SessionPersistence
+	if rule.SessionPersistence != nil {
+		sessionPersistence = &ir.SessionPersistence{
+			SessionName: *rule.SessionPersistence.SessionName,
+		}
+		if rule.SessionPersistence.Type != nil && *rule.SessionPersistence.Type == gwapiv1.HeaderBasedSessionPersistence {
+			sessionPersistence.Header = &ir.HeaderBasedSessionPersistence{}
+		} else {
```

Comment:
Does this mean when `rule.SessionPersistence.Type == nil`, we treat this as Cookie type? Should we explictly set this field?

Reply:
Yes, this is because `CookieBasedSessionPersistence` is default.
https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1/shared_types.go#L780-L781

(Maybe actually gateway API fills `Type` with `CookieBasedSessionPersistence` when empty `Type` though.)